File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int right=nums.size()-1;
        int left=0;
        
        int ans=10000;
        while(right>left){
            int mid=left+(right-left)/2;
            if(nums[mid]>nums[right]){
                left=mid+1;
            }
            else {
                right=mid;
            }
        }
        return nums[left];
    }
};

File: Data Structures & Algorithms/capacity-to-ship-packages-within-d-days/submission-0.py
class Solution:
    def shipWithinDays(self, weights: List[int], days: int) -> int:
        left = max(weights)
        right = sum(weights)
        
        def ship(capacity: int) -> bool:
            day = 1
            weight = 0
            for w in weights:
                if weight + w > capacity:
                    day += 1
                    weight = 0
                weight += w
            return day <= days
        
        while left < right:
            mid = (left + right) // 2
            if ship(mid):
                right = mid
            else:
                left = mid + 1
        return left